DateTime Item doesn't accept ZonedDateTime

This code failed:

```ruby
DateTimeItem1 << ZonedDateTime.now
```

```
13:02:38.797 [WARN ] [ab.core.model.script.actions.BusEvent] - Cannot convert '2021-03-05T13:02:38.788380+10:00[Australia/Brisbane]' to a state type which item 'DateTimeItem1' accepts: [DateTimeType, UnDefType].
```

I understand that this is actually an openhab-core problem. Do we want to intercept and correct it?

The same error occurred with #update too

No, just tested in a jython rule, and it doesn't work there either. The error message is much worse there as well:
```python
from core.rules import rule
from core.triggers import when
from java.time import ZonedDateTime

@rule("Test")
@when("Item Test received command")
def test(event):
    now = ZonedDateTime.now().toString()
    test.log.warn(now)
    events.postUpdate("TestDate", now)
```
```
2021-03-05 14:41:44.757 [WARN ] [jsr223.jython                       ] - Traceback (most recent call last):
  File "/etc/openhab/automation/lib/python/core/log.py", line 65, in wrapper
    return function(*args, **kwargs)
  File "/etc/openhab/automation/lib/python/core/rules.py", line 109, in execute
    self.callback(inputs.get('event'))
  File "<script>", line 10, in test
IllegalArgumentException: java.lang.IllegalArgumentException: The argument 'state' must not be null.
2021-03-05 14:41:44.784 [ERROR] [e.automation.internal.RuleEngineImpl] - Failed to execute rule '34238d9f-9814-4eee-be6d-05dbf7e1a1b8': Fail to execute action: 1
```

--- a/docs/usage/items/datetime.md
+++ b/docs/usage/items/datetime.md
@@ -15,10 +15,11 @@ them are extended or adapted with OpenHAB specific functionality.
 
 ##### Examples
 
-DateTime items can be updated and commanded with ruby Time objects
+DateTime items can be updated and commanded with ruby Time objects or Java ZonedDateTime objects
 
 ```ruby
 Example_DateTimeItem << Time.now
+Example_DateTimeItem << ZonedDateTime.now
 ```
 
 Math operations (+ and -) are available to make calculations with time in a few different ways

--- a/features/date_time_item.feature
+++ b/features/date_time_item.feature
@@ -99,3 +99,15 @@ Feature: date_time_item
         """
       When I deploy the rules file
       Then It should log "05:00 is between DateOne..DateTwo" within 5 seconds
+
+    Scenario: DateTime items accept ZonedDateTime
+      Given code in a rules file
+        """
+        require 'java'
+        java_import java.time.ZoneId
+        DateOne << ZonedDateTime.of(1999,12,31,0,0,0,0,ZoneId.of("UTC"))
+        """
+      When I deploy the rules file
+      Then "DateOne" should be in state "1999-12-31T00:00:00.000+0000" within 5 seconds
+
+

--- a/lib/openhab/dsl/items/date_time_item.rb
+++ b/lib/openhab/dsl/items/date_time_item.rb
@@ -2,6 +2,7 @@
 
 require 'forwardable'
 require 'time'
+require 'java'
 
 require 'openhab/dsl/items/comparable_item'
 
@@ -9,6 +10,7 @@ module OpenHAB
   module DSL
     module Items
       java_import org.openhab.core.library.items.DateTimeItem
+      java_import java.time.format.DateTimeFormatter
 
       # Adds methods to core OpenHAB DateTimeItem type to make it more natural
       # in Ruby
@@ -53,6 +55,7 @@ module OpenHAB
         # @!visibility private
         def format_type(command)
           return command.iso8601 if command.respond_to?(:iso8601)
+          return DateTimeFormatter::ISO_OFFSET_DATE_TIME.format(command) if command.is_a? java.time.ZonedDateTime
 
           super
         end

